hub/models.py

```diff
@@ -898,84 +1014,290 @@ async def fetch_many_loader(self, keys):
+                    geography_column_type__isnull=False,
+                ).all()
+            },
+        )
+
+        return loaders
 
-    async def map_one(
-        self,
-        member: Union[str, dict],
-        config: "ExternalDataSourceUpdateConfig",
-        loaders: Loaders,
-    ) -> MappedMember:
+    async def map_one(self, member: Union[str, dict], loaders: Loaders) -> MappedMember:
         """
         Match one member to a record in the data source, via ID or record.
         """
         if type(member) is str:
             member = await loaders["fetch_record"].load(member)
-        # Get postcode field from the config
-        postcode_column = config.postcode_column
-        # Get postcode from member
-        postcode = self.get_record_field(member, postcode_column)
+        update_fields = {}
         try:
-            # Get relevant config data for that postcode
-            postcode_data = await loaders["postcodesIO"].load(postcode)
+            postcode_data = None
+            if self.geography_column_type == self.PostcodesIOGeographyTypes.POSTCODE:
+                # Get postcode from member
+                postcode = self.get_record_field(member, self.geography_column)
+                # Get relevant config data for that postcode
+                postcode_data: PostcodesIOResult = await loaders["postcodesIO"].load(
+                    postcode
+                )
             # Map the fields
-            update_fields = {}
-            for mapping_dict in config.get_mapping():
+            for mapping_dict in self.get_update_mapping():
                 source = mapping_dict["source"]
-                path = mapping_dict["source_path"]
-                field = mapping_dict["destination_column"]
+                source_path = mapping_dict["source_path"]
+                destination_column = mapping_dict["destination_column"]
                 if source == "postcodes.io":
                     if postcode_data is not None:
-                        update_fields[field] = get(postcode_data, path)
+                        update_fields[destination_column] = get(
+                            postcode_data, source_path
+                        )
                 else:
-                    pass
+                    try:
+                        source_loader = loaders["source_loaders"].get(source, None)
+                        if source_loader is not None:
+                            update_fields[
+                                destination_column
+                            ] = await source_loader.load(
+                                self.EnrichmentLookup(
+                                    member_id=self.get_record_id(member),
+                                    postcode_data=postcode_data,
+                                    source_id=source,
+                                    source_path=source_path,
+                                )
+                            )
+                    except Exception:
```

Comment:
Do we want to throw something a bit harder so we can see these error in any logging we have?

Reply:
Going to add a `TODO: sentry logging` comment as a marker and then revisit when we install Sentry